src/bmson/parse.rs

```diff
@@ -0,0 +1,230 @@
+//! This is a parser for JSON.
+
+use ariadne::{Color, Label, Report, ReportKind};
+use chumsky::prelude::*;
+use serde_json::Value;
+
+use crate::diagnostics::{SimpleSource, ToAriadne};
+
+/// This is a parser for JSON.
+///
+/// Parsing from str, returning [`Value`] and error with [`Rich`] type.
+#[must_use]
+pub fn parser<'a>() -> impl Parser<'a, &'a str, Value, extra::Err<Rich<'a, char>>> {
```

Comment:
What I suggested to change is the return type of this function. Its error type is `Err<Rich>` which mixes warnings and errors. So I want to improve it into a custom error `enum` to split them out.